airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/declarative_authenticator.py

```diff
@@ -0,0 +1,13 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC
+from dataclasses import dataclass
+
+
+@dataclass
+class DeclarativeAuthenticator(ABC):
```

Comment:
This abstraction is needed so that we only attempt to expand certain authenticators during validation. It is a little bit of a concern that schema generation is coupled very closely to our exact object model which could get a little unwieldly in the future.